app/diary/sections/put_entry.py

```diff
@@ -1,39 +1,97 @@
+        error_messages.append("Bad Request: Invalid section_id")
     if diary_id is None or not isinstance(diary_id, str):
-        return {"statusCode": 400, "body": "Bad Request: Invalid diary_id"}
-    if not body:
-        return {"statusCode": 400, "body": "Bad Request: Missing request body"}
-
-    # ここに処理を書く
-    example = {
-        "diary_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
-        "date": "2023-08-31",
-        "details": "hoge.fugaの単体テストを作成する",
-        "serious": 0,
-        "created_at": "2020-01-01T00:00:00+09:00",
-        "updated_at": "2020-01-01T00:00:00+09:00",
-        "section": {
-            "section_id": 0,
-            "name": "営業課",
-            "created_at": "2020-01-01T00:00:00+09:00",
-            "updated_at": "2020-01-01T00:00:00+09:00",
+        error_messages.append("Bad Request: Invalid diary_id")
+    if details is None or not isinstance(details, str):
+        error_messages.append("Bad Request: Invalid details")
+    if serious is None or not isinstance(serious, int):
+        error_messages.append("Bad Request: Invalid serious")
+    if user_ids is None or not isinstance(user_ids, list):
+        error_messages.append("Bad Request: Invalid user_ids")
+    if len(error_messages) > 0:
+        return {
+            "statusCode": 400,
+            "body": "\n".join(error_messages),
+        }
+
+    # ユーザーの存在確認
+    try:
+        for user_id in user_ids:
+            get_item(user_table, "user_id", user_id)
+        # sectionの取得
+        section = get_item(section_table, "section_id", section_id)
+        # get diary
+        diary = get_item(section_diary_table, "diary_id", diary_id)
+
+        # section validation
+        if section["section_id"] != section_id:
+            return {
+                "statusCode": 400,
+                "body": "Bad Request: Invalid section_id",
+            }
+    except DynamoDBError as e:
+        return {"statusCode": 500, "body": f"Failed: {e}"}
+    except IndexError as e:
+        return {"statusCode": 404, "body": f"Failed: {e}"}
+
+    # update
+    update_resp = section_diary_table.update_item(
+        Key={"diary_id": diary_id},
+        UpdateExpression="set details=:d, serious=:s, user_ids=:u, updated_at=:upd",
+        ExpressionAttributeValues={
+            ":d": details,
+            ":s": serious,
+            ":u": user_ids,
+            ":upd": datetime.now().isoformat(),
         },
-        "user_ids": ["4f73ab32-21bf-47ef-a119-fa024bc2b9cc"],
-    }
+        ReturnValues="UPDATED_NEW",
+    )
+    if update_resp["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        return {
+            "statusCode": 500,
+            "body": f"Failed to update diary with diary id: {diary_id}",
+        }
+
+    diary = diary
+    diary["details"] = details
+    diary["serious"] = serious
+    diary["user_ids"] = user_ids
+    diary["updated_at"] = datetime.now().isoformat()
+    diary["section"] = section
```

Comment:
```suggestion
    diary = update_resp["Attributes"]
```
上を提案通りに直したら、これでよさそう？